content/400-pulse/250-database-setup/300-railway.mdx

```diff
@@ -8,93 +8,112 @@ toc: true
+7. After restarting your database, click on the **restart-db-then-delete-me** service and navigate to the **Settings** tab.
+8. Scroll down to the bottom and click the red **Delete service** button.
+9. You now have a Pulse-compatible PostgreSQL database running on Railway. You can now follow the instructions on the [Getting started](/pulse/getting-started) page.
+
+## Setup with template: "Prisma Pulse DB & App"
+
+This section provides step-by-step instructions for setting up a Pulse starter project with a Pulse-ready PostgreSQL database using the Railway template called: [Prisma Pulse DB & App](https://railway.app/template/pulse-starter).
+The template consists of three [services](https://docs.railway.app/reference/services):
+
+- **Pulse-Starter**: A Typescript app with a basic setup for using Prisma Pulse. You can find the repo for it [on GitHub](https://github.com/prisma/pulse-starter/tree/main).
+- **Postgres**: A PostgreSQL database that you'll be able to use with Prisma Pulse after going through the setup instructions.
+- **restart-and-delete-me**: This service contains a [script](https://github.com/prisma/pulse-railway-pg-config/blob/main/config-db.ts) that's executed once and configures the PostgreSQL database to make it usable with Prisma Pulse. Once the script has been executed, the instructions will tell you to delete this service since it won't be needed any more.
+
+1. Deploy the [template](https://railway.app/template/pulse-starter) on Railway.
+2. You'll be navigated to a page prompting configuration for both the **restart-db-then-delete-me** and **pulse-starter** service.
+   ![Configure Railway template](../images/railway-template-configure.png)
+   1. For the **restart-db-then-delete-me** service, click on the **Configure** button and then save the configuration by clicking on **Save Config**.
+   2. On the **pulse-starter** service, locate and click the **Configure** button. Within the **Environment variables** section, you'll find an option to include the `PULSE_API_KEY` environment variable. Keep the `PULSE_API_KEY` field blank for the time being. Once you've set up the database and obtained the database URL from the [Prisma Data Platform](https://console.prisma.io/), you can then add the `PULSE_API_KEY` to the **pulse-starter** service.
+   3. Click on the **Deploy** button.
+3. To set up the database and make it Pulse-ready, complete the steps outlined in the [previous section](#setup-with-template-prisma-pulse-db-only), starting from **2.**, then follow the instructions below.
+4. Go to the [Platform Console](https://console.prisma.io/) and activate Pulse in a Platform environment. Next, use the `DATABASE_URL` acquired from earlier steps to generate a `PULSE_API_KEY`.
+5. Click on the **pulse-starter** service.
+   > **Note**: You'll likely find that the build failed for the **pulse-starter** service. This is because the Prisma Pulse API key wasn't provided as the database wasn’t Prisma Pulse ready.
```

Comment:
```suggestion
   > **Note**: You'll likely find that the build failed for the **pulse-starter** service. This is because the Prisma Pulse API key wasn't provided. Do not worry, this is to be expected and not a problem for the next steps.
```